go-controller/pkg/ovn/ovn.go

```diff
@@ -70,17 +72,14 @@ type namespaceInfo struct {
 
 	// addressSet is an address set object that holds the IP addresses
 	// of all pods in the namespace.
-	addressSet AddressSet
+	addressSet address_set.AddressSet
```

Comment:
What if we just moved this to a `map[string]*address_set.AddressSet` so that we wouldn't have to go through `nsinfo` in order to get the `AddressSet` for a namespace

Reply:
at least for now we don't have to do anything like that. Since the namespace addressSet is a string in a match expression it doesn't have to actually exist, the generation of the name is known I will fix that up

Reply:
Hmm we could decouple the namespace creation from the creation of the addressSet. The AddIP function could check for the existence of the addres_set and if it is nil create it as well as every delete will check if the address_set is empty and if it is delete it but I would like some more input on that @dcbw do you have any input on that? AddIPs would also need to take the name of the address_set, which in the case of a namespace address_set will always be known, but is there any other issues you might see from this change?